test/com/twu/biblioteca/MainMenuTest.java

```diff
@@ -74,4 +81,17 @@ public void shouldDisplayInvalidMessageWhenInvalidOptionSelected() {
                         System.getProperty("line.separator"),
                 output.toString());
     }
+
+    @Test
+    public void shouldCloseApplicationWhenQuitOptionSelected() {
+        MainMenu mainMenu = new MainMenu();
+
+        System.setIn(new ByteArrayInputStream(("2").getBytes()));
+
+        mainMenu.showOptions();
+
+        exit.expectSystemExit();
```

Comment:
Wow! This is a new on to me... I didn't know this was possible.  I suspect what this is doing is mocking out the System class and intercepting the call to "exit".

Reply:
Yes, I had to use a 3rd party library to implement a rule for SystemExit.

https://stefanbirkner.github.io/system-rules/index.html

It was the only way I could out to test the quit option though I suspect there must be other out of the box ways to do it.

Reply:
The only other way I know of is doing a system-level test where you start the application in the test (e.g., "java ..."), feed it input and then test that it has exited.  But that approach would be beyond the scope of this work and something I wouldn't normally test in most cases